pkg/sql/pg_catalog.go

```diff
@@ -2775,6 +2774,53 @@ CREATE TABLE pg_catalog.pg_type (
 				})
 			})
 	},
+	indexes: []virtualIndex{
+		{
+			partial: false,
+			populate: func(ctx context.Context, constraint tree.Datum, p *planner, db *sqlbase.ImmutableDatabaseDescriptor,
+				addRow func(...tree.Datum) error) (bool, error) {
+
+				h := makeOidHasher()
+				nspOid := h.NamespaceOid(db, pgCatalogName)
+				coid := tree.MustBeDOid(constraint)
+				ooid := oid.Oid(int(coid.DInt))
+
+				// Check if it is a predefined type
+				typ, ok := types.OidToType[ooid]
+				if ok {
+					if err := addPGTypeRow(h, nspOid, typ, addRow); err != nil {
+						return false, err
+					}
+					return true, nil
+				}
+
+				// Check if it is a user defined type
+				descs, err := p.Tables().GetAllDescriptors(ctx, p.txn)
```

Comment:
This is an index lookup, so we don't want to have to scan all of the descriptors! You should instead use `sqlbase.GetTypeDescFromID` and leave a TODO for me (due to some related work going on): `TODO (rohany): This access should go through the desc.Collection`.

Reply:
Done